net: improve code documentation for dns seed behaviour

ajtowns opened the pull request:
Some better internal documentation post #16939

ajtowns committed:
net: improve code documentation for dns seed behaviour

laanwj commented on src/net.cpp:
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
+/** How long to delay before querying DNS seeds
+ *
+ * If we have more than THRESHOLD entries in addrman, then it's likely
+ * that we got those addresses from having previously connected to the p2p
```
capitalize P2P please

laanwj commented on src/net.cpp:
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
+ * little longer trying to connect to known peers before querying the
+ * DNS seeds.
+ */
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11}; // 11sec
```
the `11sec` comment is needless here, the constant (using `std::chrono` type is self-documenting)

laanwj commented on src/net.cpp:
```diff
@@ -51,6 +51,19 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
+ * DNS seeds.
+ */
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11}; // 11sec
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_MANY_PEERS{300}; // 5min
```
same but std::chrono::minutes(5)` ?

fanquake commented:
Concept ACK. ecc86b9d514c6d67766eb2dd76f745ebfa36238a looks pretty good. Can you rebase and mark as ready for review?

hebasto commented:
Concept ACK.

MarcoFalke commented on src/net.cpp:
```diff
@@ -52,10 +52,17 @@ static constexpr int DUMP_PEERS_INTERVAL = 15 * 60;
-static constexpr std::chrono::seconds DNSSEEDS_DELAY_MANY_PEERS{300}; // 5min
-static constexpr int DNSSEEDS_DELAY_PEER_THRESHOLD = 1000; // "many" vs "few" peers -- you should only get this many if you've been on the live network
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11};
+static constexpr std::chrono::minutes DNSSEEDS_DELAY_MANY_PEERS{5};
```
ACK on this change

naumenkogs commented:
ACK ecc86b9

ajtowns commented:
Rebased, un-drafted PR, added some comments reflecting [review by @Sjors](https://github.com/bitcoin/bitcoin/pull/16939#pullrequestreview-420310676)

fanquake commented on src/net.cpp:
```diff
@@ -53,10 +53,17 @@ static constexpr std::chrono::minutes DUMP_PEERS_INTERVAL{15};
-static constexpr int DNSSEEDS_DELAY_PEER_THRESHOLD = 1000; // "many" vs "few" peers -- you should only get this many if you've been on the live network
+static constexpr std::chrono::seconds DNSSEEDS_DELAY_FEW_PEERS{11};
+static constexpr std::chrono::minutes DNSSEEDS_DELAY_MANY_PEERS{5};
+static constexpr int DNSSEEDS_DELAY_PEER_THRESHOLD = 1000; // "many" vs "few" peers
```
nit: `"many" vs "few" peers` seems redundant now that there is an explainer above.

fanquake reviewed (approved):
ACK 5cb7ee67a5c0e5da39eb698b64d23722fb2f7b3e - thanks for following up.

hebasto commented on src/net.cpp:
```diff
@@ -53,10 +53,17 @@ static constexpr std::chrono::minutes DUMP_PEERS_INTERVAL{15};
 
 /** How long to delay before querying DNS seeds
+ *
+ * If we have more than THRESHOLD entries in addrman, then it's likely
```
nit: maybe
```suggestion
 * If we have more than PEER_THRESHOLD entries in addrman, then it's likely
```
?

hebasto reviewed (commented):
ACK 5cb7ee67a5c0e5da39eb698b64d23722fb2f7b3e

ariard commented on src/net.cpp:
```diff
@@ -1797,6 +1809,9 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
         // Add seed nodes if DNS seeds are all down (an infrastructure attack?).
+        // Note that we only do this if we started with an empty peers.dat,
+        // (in which case we will query DNS seeds immediately) *and* the DNS
+        // seeds have not returned any results.
```
I don't think calling those peers "fixed seeds" matches with DNS seeds meaning. Contrary to DNS seeds open as `fOneShot=true`, we open regular connections to them, and not querying a `ADDR` and then disconnecting ?

naumenkogs replied:
I agree it's a bit confusing that we call both of them seeds, but the problem is not with this PR. This PR just adapts to the existing naming.
Are you suggesting renaming fixed seeds to hardcoded nodes or something?
That alone would require a separate PR.

ariard replied:
Yes I'm suggesting renaming them with a different names to mark difference of mechanism in a separate PR.

fanquake replied:
I'm going to merge this now. @ariard please open a followup for this (and if you want the other nits).

ariard reviewed (commented):
ACK 5cb7ee6

naumenkogs commented:
ACK 5cb7ee6
